packages/svelte/src/internal/client/dom/elements/transitions.js

```diff
@@ -311,6 +311,12 @@ function animate(element, options, counterpart, t2, callback) {
 				if (animation.startTime !== null && animation.currentTime !== null) {
 					throw e;
 				}
+				// If we have an aborted request and the element is detached from the DOM, then that
+				// means that it's likely that the animation was paused upon the DOM element being
+				// removed. This won't trigger the finish logic, so we need to ensure the callback is fired.
+				if (!element.isConnected) {
+					callback?.();
+				}
```

Comment:
is there a test we can add around this stuff? if i comment it out, or comment the `if` out but keep the callback, nothing fails.

the reason i'm harping on this is that words like 'likely' suggest non-determinism — given how convoluted transitions are in general, i think we need to be hyper-specific about this stuff if we're to avoid regressions. (related: not sure what 'aborted request' means here?)

Reply:
I'm not sure how this is possible to test to be honest. We don't capture any of this stuff in our hacky polyfill because the error that is caught is not something JSDOM knows about, nor does our polyfill. It also seems browser engine specific too.

I honestly have little faith in the web animations spec, it's really brittle. Possibly one of the most troublesome web APIs I've ever had to work with too. I'd maybe argue that we should probably drop the Promise based API entirely, as it's just flat out unreliable.